docs/index/secondary-index-guide.md

```diff
@@ -0,0 +1,155 @@
+  So first level of pruning will be done on SI table as it stores blocklets and main table/parent
+  table pruning will be based on the SI output, which helps in giving the faster query results with
+  better pruning.
+
+  For instance, main table called **sales** which is defined as
+
+  ```
+  CREATE TABLE sales (
+    order_time timestamp,
+    user_id string,
+    sex string,
+    country string,
+    quantity int,
+    price bigint)
+  STORED AS carbondata
+  ```
+
+  User can create SI table using the Create Index DDL
+
+  ```
+  CREATE INDEX index_sales
+  ON TABLE sales(user_id)
+  AS
+  'carbondata'
+  TBLPROPERTIES('table_blocksize'='1')
+  ```
+**NOTE**:
+ * Secondary Index tables are registered with hive and stored n HiveSERDEPROPERTIES s json formatted
+ value. Maximum characters supported for SERDEPROPERTIES by hive is 4000 characters which cannot be
+ changed.
+ 
+ 
+#### How SI tables are selected
+
+When a user filter query is submitted, during query planning phase, CarbonData with help of
+CarbonSITransformationRule, checkes if there are any index tables present on the filter column of
+query. If there are any, then filter query plan will be transformed such a way that, execution will
+first hit the corresponding SI table and give input to main table for further pruning.
+
+
+For the main table **sales** and SI table  **index_sales** created above, following queries
+```
+SELECT country, sex from sales where user_id = 'xxx'
+
+SELECT country, sex from sales where user_id = 'xxx' and country = 'INDIA'
+```
+
+will be transformed by CarbonData's CarbonSITransformationRule to query against SI table
+**index_sales** first which will be input to the main table **sales**
+
+
+## Loading data
+
+### Loading data to Secondary Index table(s).
+
+*case1:* When SI table is created and the main table does not have any data. In this case every
+consecutive load will load to SI table once main table data load is finished.
+
+*case2:* When SI table is created and main table already contains some data, then SI creation will
+also load to SI table with same number of segments as main table. There after, consecutive load to
+main table will load to SI table also.
+
+ **NOTE**:
+ * In case of any failure to SI table, then we make the SI table disable by setting a hive serde
+ property and subsequent load to main table will also load to SI with failed load data also and
+ make it enable and available for query.
+
+## Querying data
+Direct query can be made on SI tables to see the data present in position reference columns.
+When a filter query is fired, if the filter column is a secondary index column, then plan is
+transformed accordingly to hit SI table first to make better pruning with main table and in turn
+helps for faster query results.
+
+User can verify whether a query can leverage SI table or not by executing `EXPLAIN`
+command, which will show the transformed logical plan, and thus user can check whether SI table
+table is selected.
```

Comment:
Can add documentation for queries which do not push down to SI

Reply:
thats already explained right, if no filester on SI, then it will not push down no need to add more queries as eamples

Reply:
with filter on SI and donotPushToSI. For expressions like NOT, ISNOTNULL etc

Reply:
i think, we have many things like that like, filter combinations of or , and so, better to avoid these kind in user documents , it will be better to mention in developer document